x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MlMetadata.java

```diff
@@ -287,7 +287,7 @@ public Builder deleteJob(String jobId, PersistentTasksCustomMetaData tasks) {
             if (job == null) {
                 throw new ResourceNotFoundException("job [" + jobId + "] does not exist");
             }
-            if (job.isDeleted() == false) {
+            if (job.isDeleting() == false) {
```

Comment:
So, what happens when `deleteJob` gets called twice in a row while a job is in the `deleting` status? On the second call, would the error not be `ResourceNotFoundException("job [" + jobId + "] does not exist");` Or is the `private final SortedMap<String, Job> jobs;` repopulated somewhere?

Reply:
👍